practice/completer/site.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+/**
+ * Completer object contains completions.
+ *
+ * completions will have keys as the completion and values as weights.
+ */
+function Completer() {
+  this.completions = {};
+}
+
+/**
+ * modifyCompleter has these methods:
+ *
+ * addCompletion() adds a completion to the completions container (in-place)
+ * removeCompletion() removes a completion from the container (in-place)
+ * complete() returns an array of completions, highest weighted first
+ * selectCompletion() increases the weight of a given completion (in-place)
+ *
+ */
+var modifyCompleter = {
+  addCompletion: function(str) {
+    this.completions[str] = 1;
```

Comment:
One thing to consider is if the user inputted a duplicate value, ```completions[str]``` would get reset to 1, which might be undesirable!